### PR TITLE
Remove reliance on brentru/ci-arduino

### PR DESCRIPTION
**Pull request by brentru**
- Remove brentru/ci-arduino, use adafruit/ci-arduino instead
- Match debug build targets to new names proposed in https://github.com/adafruit/ci-arduino/pull/174/

### CODE REVIEW

File: .github/workflows/build-clang-doxy.yml [hunk 1]
```diff
@@ -389,10 +389,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        arduino-platform: ["feather_esp32s2_dev", "feather_esp32s2_tft_dev",
-                           "feather_esp32s2_reverse_tft_dev", "feather_esp32s3_dev",
-                           "feather_esp32s3_4mbflash_2mbpsram_dev", "feather_esp32s3_tft_dev",
-                           "feather_esp32s3_reverse_tft_dev"]
+        arduino-platform: ["feather_esp32s2_debug", "feather_esp32s2_tft_debug",
+                           "feather_esp32s3_debug", "feather_esp32s3_4mbflash_2mbpsram_dev", "feather_esp32s3_tft_debug",
```
Comment (tyeth) [inline .github/workflows/build-clang-doxy.yml:393]: Looks like _dev should be _debug for the 2mb psram

**Review by tyeth — CHANGES_REQUESTED** [review]
Not quite sure if we need skip files for the no-FS sketch for the debug builds

File: .github/workflows/build-clang-doxy.yml [hunk 2]
```diff
@@ -389,10 +389,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        arduino-platform: ["feather_esp32s2_dev", "feather_esp32s2_tft_dev",
-                           "feather_esp32s2_reverse_tft_dev", "feather_esp32s3_dev",
-                           "feather_esp32s3_4mbflash_2mbpsram_dev", "feather_esp32s3_tft_dev",
-                           "feather_esp32s3_reverse_tft_dev"]
+        arduino-platform: ["feather_esp32s2_debug", "feather_esp32s2_tft_debug",
+                           "feather_esp32s3_debug", "feather_esp32s3_4mbflash_2mbpsram_debug", "feather_esp32s3_tft_debug",
+                           "feather_esp32s3_tft_debug"]
```
Comment (tyeth) [inline .github/workflows/build-clang-doxy.yml:394]: Just spotted this is a repeat. I think the reverses got dropped / amended

Comment (brentru) [reply .github/workflows/build-clang-doxy.yml:394]: They got removed in the PR to ci-arduino, I wanted two builds per feather (one with tft, one without).

**Pull request merged**